ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/InMemoryKVParticipantState.scala

```diff
@@ -35,18 +34,24 @@ import org.slf4j.LoggerFactory
 
 import scala.collection.JavaConverters._
 import scala.collection.breakOut
-import scala.concurrent.duration._
+import scala.concurrent.duration.DurationInt
 import scala.concurrent.{Await, ExecutionContext, Future}
 import scala.util.Try
 
 object InMemoryKVParticipantState {
 
+  /** Get a new record time for the ledger from the system clock.
+    * Public for use from integration tests.
+    */
+  private def now(): Timestamp =
```

Comment:
Would it make sense to rather let constructor callers optionally pass a `() => Timestamp` so that they can read it?

Reply:
Maybe passing a `Clock` could make sense in this case, depending on what the callers have to expect.

Reply:
I just moved and renamed this function to fix a couple of warnings. Don't want to improve upon this code because it's going to be deleted.